home/import_content_pages.py

```diff
@@ -189,10 +190,11 @@ def add_go_to_page_buttons(self) -> None:
                             f"page '{slug}'",
                             row.row_num,
                         )
-                    page.whatsapp_body[message_index].value["buttons"].append(
-                        ("go_to_page", {"page": related_page, "title": title})
+                    page.whatsapp_body[message_index].value["buttons"].insert(
+                        button["index"],
+                        ("go_to_page", {"page": related_page, "title": title}),
                     )
-            page.save_revision().publish()
+            page.save()
```

Comment:
What implications does this change have? Will the updates to the buttons be unpublished?

Reply:
If I'm understanding the docs, it will update the currently published page. This is my understanding:
- The `Page` object in the database contains the current live/published content
- The `Revision` object contains the history of previous versions of the page
- So if you do a `save_revision`, it creates a new revision (a new point in the history). Because it's newer than the current page, it's seen as unpublished changes. If you then call `publish` on that, it copies whatever is in that revision onto the `Page`, making it live.

So if you just do a `save_revision`, then we'll have unpublished buttons until we call `publish` on that revision. If we do a `save`, it should update the published page with new buttons.

I think that's okay, unless there's something I'm not thinking about?

Reply:
If you're happy with that, I am too -- I've never been able to wrap my head around wagtail's behaviour. Do we support importing unpublished content? If so, do we have tests for the behaviour there?

Reply:
No we don't support that. Export and import only works on published content, we assume that anything you're importing you want published when the import is done.

We also don't import/export revision history, so by doing a "purge" import you do lose that history (there's also things like comments and moderation review that we don't import/export).